### PR TITLE
Sync labels 

### PR DESCRIPTION
**Pull request by langleyd**
## What's in this PR?
- Add sync labels script to sync with element-meta labels and merge with local labels
- Add local labels file for compound specific labels

## TODO
- [x] Somebody with access to the repo secrets to confirm `ELEMENT_BOT_TOKEN` exists.

### CODE REVIEW

File: .github/labels.yml
Comment (pixlwave) [inline .github/labels.yml:1]: Not sure what this is doing, but I trust it is correct 👍

**Review by pixlwave — APPROVED** [review]
LGTM, thanks

**Comment by pixlwave** [comment]
@langleyd Errr, ok it failed to run: https://github.com/element-hq/compound/actions/runs/11176351416/job/31069524078

Also, looking at that run, I wonder, do we actually want to sync labels with Compound? E.g. things like `Rust: Expert` don't particularly seem useful on this repo.

**Comment by langleyd** [comment]
@pixlwave Looks like [GH gives a 404 for insufficient privileges](https://github.com/orgs/community/discussions/24604#discussioncomment-3244636), so I'm guessing the `ELEMENT_BOT_TOKEN` could be invalid. I can have IT make sure it's up to date.

sync-labels merges labels will element-meta and the local labels.yml file. It means we can maintain common labels across repos from a single place(element-meta), like the `T-` `O-` `S-` `A-` if we want to have common process/consistency. There are defo some in meta that need to be cleaned up(like old teams), but at least it can be done in one place.

As you mention, it comes at the expense of some extraneous labels, but they can just be ignored? WDYT?

**Comment by pixlwave** [comment]
> I can have IT make sure it's up to date.

👍

> There are defo some in meta that need to be cleaned up(like old teams)

Right, got it. I guess I had assumed things like team and `A-` labels wouldn't be common, but makes sense to me, thanks for explaining.

**Pull request merged**